src/node/rpc/frontend.h

```diff
@@ -479,7 +490,9 @@ namespace ccf
 
           case HandlerRegistry::MayWrite:
           {
-            if (!ctx->read_only_hint)
+            const auto read_only_it =
+              ctx->get_request_header(http::headers::CCF_READ_ONLY);
+            if (!read_only_it.has_value() || (read_only_it.value() != "true"))
```

Comment:
This is now a header rather than a field in the top-level JSON-RPC. I believe this works, but it looks like we've lost all of our `MayWrite` handlers so this isn't actually tested.